### PR TITLE
Decouple specPreview from Shelf

### LINKED ISSUE #505: Field added to the encoding preview cannot be removed
When hovering over the wildcard field in specified views, the field added to the encoding preview cannot be removed. At the same time, it triggers all plots to rerender.
bug

### PR DESCRIPTION
**Pull request by starry97**
Fix #505

### CODE REVIEW

File: src/actions/shelf-preview.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import {FacetedCompositeUnitSpec} from 'vega-lite/build/src/spec';
+import {PlainReduxAction, ReduxAction} from './redux-action';
+
+export type ShelfPreviewAction = ShelfSpecPreview | ShelfSpecPreviewDisable;
+
+export const SHELF_SPEC_PREVIEW = 'SHELF_SPEC_PREVIEW';
```
Comment (kanitw) [inline src/actions/shelf-preview.ts]: Maybe SHELF_PREVIEW_SPEC?  (So the state remains the prefix.)

**Comment by starry97** [comment]
@kanitw Yes, this fixes #505

**Comment by felixcodes** [comment]
@kanitw @starry97 
Just an FYI: we'll have to decouple this a step further later as the preview is added to the undo-redo stack.

**Comment by kanitw** [comment]
@felixcodes Good catch.  Maybe add a TODO item to the decouple issue?

**Pull request merged**